examples/server/webui/src/utils/llama-vscode.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+import { MessageExtraTextFile } from './types';
+
+// Extra context when using llama.cpp WebUI from llama-vscode, inside an iframe
+// Ref: https://github.com/ggml-org/llama.cpp/pull/11940
+
+interface SetTextEvData {
+  text: string;
+  context: string;
+}
+
+/**
+ * To test it:
+ * window.postMessage({ command: 'setText', text: 'Spot the syntax error', context: 'def test()\n  return 123' }, '*');
+ */
+
+export const useVSCodeContext = (
+  inputRef: React.RefObject<HTMLTextAreaElement>,
+  setInputMsg: (text: string) => void
+) => {
+  const [extraContext, setExtraContext] = useState<MessageExtraTextFile | null>(
+    null
+  );
+  console.log({ extraContext });
+
+  // Accept setText message from a parent window and set inputMsg and extraContext
+  useEffect(() => {
+    const handleMessage = (event: MessageEvent) => {
+      if (event.data?.command === 'setText') {
+        const data: SetTextEvData = event.data;
+        setInputMsg(data?.text);
+        setExtraContext({
+          type: 'textFile',
+          name: '(context)', // TODO: set filename
```

Comment:
@igardev This is nice-to-have, but can you set the file name inside the event data sent from vscode?

Reply:
I removed the file name then